### PR TITLE
Fix #522: Change the field type 'matches' of 'FileOperationPattern' to be String

### LINKED ISSUE #522: No way to call FileOperationPattern.setMatches() to only match files or folders
When i want to register `FileOperationsServerCapabilities` to enable willRenameFiles support, i found it's not possible to create a FileOperationPattern to only match files or folders.

The setter of the field `FileOperationPatternKind matches` asks for an instance of FileOperationPatternKind. See https://github.com/eclipse/lsp4j/blob/4901031aa9fb277711b68a2a8dc0a234fab41b5f/org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend#L7036-L7058

But the class `FileOperationPatternKind` hides its constructor and no way to create an instance for it. See 
https://github.com/eclipse/lsp4j/blob/4901031aa9fb277711b68a2a8dc0a234fab41b5f/org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/FileOperationPatternKind.java#L20-L33

We should make the matches field to be String type, and accept FileOperationPatternKind.File and FileOperationPatternKind.Folder as value.

A fix could be:
```
class FileOperationPattern {

    	/**
	 * Whether to match files or folders with this pattern.
	 *
	 * The match kind could be <code>FileOperationPatternKind.File</code>, <code>FileOperationPatternKind.Folder</code>.
	 *
	 * Matches both if undefined.
	 */
	String matches

}
```

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

### CODE REVIEW

**Comment by KamasamaK** [comment]
@eclipse-lsp4j-bot ok to test

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend [hunk 1]
```diff
@@ -7053,9 +7053,12 @@ class FileOperationPattern {
 	/**
 	 * Whether to match files or folders with this pattern.
 	 *
+	 * The match kind could be <code>FileOperationPatternKind.File</code>,
+	 * <code>FileOperationPatternKind.Folder</code> and undefined.
+	 *
 	 * Matches both if undefined.
 	 */
-	FileOperationPatternKind matches
```
Comment (cdietrich) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:7058]: is this a general new pattern how we deal with enums?

Comment (cdietrich) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:7058]: should we also remove the enum then?

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend [hunk 2]
```diff
@@ -7053,9 +7053,12 @@ class FileOperationPattern {
 	/**
 	 * Whether to match files or folders with this pattern.
 	 *
+	 * The match kind could be <code>FileOperationPatternKind.File</code>,
```
Comment (KamasamaK) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:7056]: Use `{@link FileOperationPatternKind}` to reference the source of values.

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend (hunk 1 above)
Comment (KamasamaK) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:7058]: Declaring it as a `String` is the standard pattern used in this project when the enum is a string. The *Kind class is just used as a helper in this case. It was simply a mistake to not make it a `String` initially.

Comment (KamasamaK) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:7058]: See #283

Comment (testforstephen) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:7058]: I just quickly search the classes with `*Kind`, looks like we use enum for number type (such as `CompletionItemKind`), and string literal for string type (such as `FoldingRangeKind`).

`FileOperationPatternKind` accepts `file` and `folder` as value, so using string literals look ok.

Comment (testforstephen) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend:7058]: thank @KamasamaK for clarification.

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend (hunk 2 above)
Comment (testforstephen) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend]: Nice suggestion!

**Review by KamasamaK — APPROVED** [review]

**Pull request merged**